File: research4/python/single_thread.py
import random
import sys

SAMPLES = 50_000_000


def main() -> None:
    inside = 0

    for _ in range(SAMPLES):
        x = random.random()
        y = random.random()
        if x * x + y * y <= 1.0:
            inside += 1

    pi_estimate = 4.0 * inside / SAMPLES
    print(f"Samples: {SAMPLES} | Pi estimate: {pi_estimate:.6f} | Inside: {inside}")


if __name__ == "__main__":
    main()

File: research4/rust/src/bin/single_thread.rs
use rand::Rng;

const SAMPLES: u64 = 50_000_000;

fn main() {
    let mut rng = rand::rng();
    let mut inside: u64 = 0;

    for _ in 0..SAMPLES {
        let x: f64 = rng.random();
        let y: f64 = rng.random();
        if x * x + y * y <= 1.0 {
            inside += 1;
        }
    }

    let pi_estimate = 4.0 * inside as f64 / SAMPLES as f64;
    println!(
        "Samples: {} | Pi estimate: {:.6} | Inside: {}",
        SAMPLES, pi_estimate, inside
    );
}